packages/ensnode-sdk/src/ens/metadata-service.ts

```diff
@@ -1,31 +1,41 @@
 import type { ENSNamespaceId } from "@ensnode/datasources";
 import { ENSNamespaceIds } from "@ensnode/datasources";
 
+import type { BrowserSupportedAssetUrl } from "../shared/url";
+import { toBrowserSupportedUrl } from "../shared/url";
 import type { Name } from "./types";
 
 /**
- * Build the avatar image URL for a name on the given ENS Namespace that (once fetched) would
- * load the avatar image for the given name from the ENS Metadata Service
+ * Builds a browser-supported avatar image URL for an ENS name using the ENS Metadata Service
  * (https://metadata.ens.domains/docs).
  *
- * The returned URL is dynamically built based on the provided ENS namespace. Not all ENS
- * namespaces are supported by the ENS Metadata Service. Therefore, the returned URL may
- * be null.
+ * ENS avatar text records can specify URLs using protocols that browsers don't natively support
+ * for direct image rendering, such as ipfs://, ar://, or NFT URIs (eip155:). The ENS Metadata
+ * Service acts as a proxy to resolve these non-browser-supported protocols and serve the avatar
+ * images via standard HTTP/HTTPS, making them directly usable in <img> tags and other browser
+ * contexts.
+ *
+ * The returned URL uses the BrowserSupportedAssetUrl type, indicating it's safe to use directly
+ * in browsers without additional protocol handling.
  *
  * @param {Name} name - ENS name to build the avatar image URL for
  * @param {ENSNamespaceId} namespaceId - ENS Namespace identifier
- * @returns avatar image URL for the name on the given ENS Namespace, or null if the given
- *          ENS namespace is not supported by the ENS Metadata Service
+ * @returns A browser-supported avatar image URL for the name on the given ENS Namespace, or null
+ *          if the given ENS namespace is not supported by the ENS Metadata Service
  */
 export function buildEnsMetadataServiceAvatarUrl(
   name: Name,
   namespaceId: ENSNamespaceId,
-): URL | null {
+): BrowserSupportedAssetUrl | null {
   switch (namespaceId) {
     case ENSNamespaceIds.Mainnet:
-      return new URL(name, `https://metadata.ens.domains/mainnet/avatar/`);
+      return toBrowserSupportedUrl(
+        new URL(name, `https://metadata.ens.domains/mainnet/avatar/`).toString(),
```

Comment:
```suggestion
      return toBrowserSupportedUrl(`https://metadata.ens.domains/mainnet/avatar/{$encodeURIComponent(name)}`),
```

Seems more clean. wdyt?